src/include/miopen/tensor_layout.hpp

```diff
@@ -69,6 +69,42 @@ void tensor_layout_to_strides(const std::vector<T>& len,
                    });
 }
 
+template <typename T>
+void tensor_layout_to_strides_vect(const std::vector<T>& len,
```

Comment:
Perhaps merge this function with the non-vectorized version since this function can subsume the other one